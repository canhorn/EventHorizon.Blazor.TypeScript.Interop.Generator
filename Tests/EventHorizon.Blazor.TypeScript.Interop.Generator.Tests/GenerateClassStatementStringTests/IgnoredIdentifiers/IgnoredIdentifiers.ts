declare module Examples {
  export interface CameraInputsMap<T> {}
  export class DeepImmutable<T> {}

  export interface ExampleClass<V extends CameraInputsMap<I>, I> {
    property: string;
    get getProperty(): string;
    set setProperty(value: string);
    propertySignature: string;

    addToRef<R extends I>(other: DeepImmutable<I>, result: R): R;
    subtractFromFloatsToRef<R extends I>(...args: [...Array<V>, R]): R;
    method(): void;
  }
}
