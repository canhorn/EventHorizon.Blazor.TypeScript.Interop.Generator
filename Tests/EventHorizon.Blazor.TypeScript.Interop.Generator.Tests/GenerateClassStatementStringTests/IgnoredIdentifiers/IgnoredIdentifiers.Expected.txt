/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    public interface ExampleClass<V, I> : ICachedEntity { }
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClassCachedEntity<CachedEntity, CachedEntity>>))]
    public class ExampleClassCachedEntity<V, I> : CachedEntityObject, ExampleClass<V, I>
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClassCachedEntity() : base() { }

        public ExampleClassCachedEntity(
            ICachedEntity entity
        ) : base(entity)
        {
        }


        #endregion

        #region Methods

        #endregion
    }
}